validator/project_validator.go

```diff
@@ -1570,7 +1577,7 @@ func dependencyMustRun(target model.TVPair, current model.TVPair, depReqs depend
 	if depReqs.requireOnNonPatches && (taskUnit.SkipOnNonPatchBuild() || taskUnit.SkipOnNonGitTagBuild()) {
 		return false, nil
 	}
-	if depReqs.requireOnGitTag && (taskUnit.SkipOnNonPatchBuild()) {
+	if depReqs.requireOnGitTag && (taskUnit.SkipOnNonPatchBuild() || taskUnit.SkipOnGitTagBuild()) {
```

Comment:
On L1521, it seems like `requireOnGitTag` should also require that `!SkipOnGitTagBuild()`.

Reply:
good catch!